table/queries/merged_reduced_scans.sql

```diff
@@ -107,6 +111,13 @@ CREATE TEMP FUNCTION ClassifyError(error STRING,
   END
 );
 
+# Returns "abc" from the "Server: abc" header if it exists. Otherwise empty string.
+CREATE TEMP FUNCTION ExtractServerHeader(received_headers ARRAY<STRING>) AS (
+  IF(CONTAINS_SUBSTR(ARRAY_TO_STRING(received_headers, ','), "Server:"),
```

Comment:
I believe there's no need to concatenate here. That may make the query significantly slower and harder to read.

You should be able to do something like 
```sql
select REGEXP_EXTRACT(header, "Server: (.*?)")
from UNNEST(received_headers) as header
where STARTS_WITH(header, "Server: ") LIMIT 1
```

Example: http://google3/jigsaw/firehook/metrics/update_intra_windowed_stats.sql;l=4;rcl=323798496

See https://cloud.google.com/bigquery/docs/reference/standard-sql/arrays#scanning_for_values_that_satisfy_a_condition

Reply:
done